server/src/main/java/org/candlepin/controller/ContentManager.java

```diff
@@ -0,0 +1,331 @@
+     * The content instance returned by this method is not guaranteed to be the same instance passed
+     * in. As such, once this method has been called, callers should only use the instance output by
+     * this method.
+     *
+     * @param entity
+     *  The content entity to update
+     *
+     * @param owner
+     *  The owner for which to update the content
+     *
+     * @param regenerateEntitlementCerts
+     *  Whether or not changes made to the content should trigger the regeneration of entitlement
+     *  certificates for affected consumers
+     *
+     * @throws IllegalStateException
+     *  if this method is called with an entity does not exist in the backing database for the given
+     *  owner
+     *
+     * @throws NullPointerException
+     *  if the provided content entity is null
+     *
+     * @return
+     *  the updated content entity, or a new content entity
+     */
+    @Transactional
+    public Content updateContent(Content entity, Owner owner, boolean regenerateEntitlementCerts) {
+        log.debug("Applying content update for org: {}, {}", entity, owner);
+
+        if (entity == null) {
+            throw new NullPointerException("entity");
+        }
+
+        // This has to fetch a new instance, or we'll be unable to compare the objects
+        Content existing = this.contentCurator.lookupById(owner, entity.getId());
+
+        if (existing == null) {
+            // If we're doing an exclusive update, this should be an error condition
+            throw new IllegalStateException("Content has not yet been created");
+        }
+
+        if (existing == entity) {
+            // Nothing to do, really. The caller likely intends for the changes to be persisted, so
+            // we can do that for them.
+            return this.contentCurator.merge(entity);
+        }
+
+        // Check for newer versions of the same content. We want to try to dedupe as much data as we
+        // can, and if we have a newer version of the content (which matches the version provided by
+        // the caller), we can just point the given orgs to the new content instead of giving them
+        // their own version.
+        // This is probably going to be a very expensive operation, though.
+        List<Content> alternateVersions = this.contentCurator.getContentByVersion(
+            entity.getId(), entity.hashCode()
+        );
+
+        for (Content alt : alternateVersions) {
+            if (alt.equals(entity)) {
+                alt.addOwner(owner);
+                this.contentCurator.merge(alt);
+
+                // Make sure every product using the old version/entity are updated to use the new one
+                List<Product> affectedProducts = this.productCurator.getProductsWithContent(
+                    owner, Arrays.asList(entity.getId())
+                );
+
+                for (Product product : affectedProducts) {
+                    product = (Product) product.clone();
```

Comment:
I am not completely sure but it seems to me that the part of the update content that modifies affectedProducts is not covered by any spec test - no spec test has product with content that is being converged.